README.md

```diff
@@ -72,6 +73,184 @@ $myEntity = Stub::extends($myEntity, ['property1' => 'value']);
 self::assertSame('Hello world!', $myEntity->salute());
 ```
 
+### Test Checks
+
+To run test checks over your test suite, you can for example create umbrella test that allows you to configure and run desired checks.
+Data provider yields checks encapsulated in factory functions in order to allow to pass current test instance and isolate the run. 
+
+```php
+<?php
+
+use Cdn77\TestUtils\TestCheck\TestCheck;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group       integration
+ * @testedClass none
+ */
+final class SuiteComplianceTest extends TestCaseBase
```

Comment:
Now mi to přijde dost komplikovaný ... co mě napadá je this:
- tuhle třídu mít jako součást tohoto repa, a providerChecks mít jako abstract metodu
- TestCheck run metoda by suite a testFilePathNames příjmala v run metodě
- User by ve svojí třídě "SuiteComplianceTest" extendoval tu base třídu, a definoval jenom provider ... kde by navíc předával jen věci které potřebuje on, ne ty "base" věci jako je testCase a cesty k testům

Reply:
> Now mi to přijde dost komplikovaný
> User by ve svojí třídě "SuiteComplianceTest" extendoval tu base třídu

Takze usetrime dva radky, tzn. volani run metody nad checkem, na ukor flexibility, protoze tam narveme inheritance hell.

Takze napr. tenhle test uz nepujde spustit pod vlastni testcase. coz je mimochodem ta vec, kterou tu kontrolujeme, ze kazdej test dela.

Vysvetli pls co je na tom presne komplikovany

> jen věci které potřebuje on, ne ty "base" věci jako je testCase a cesty k testů
to nechapu, jaky veci tu jsou navic? tomu checku musis predat files, ktery chces checknout. Udelal jsem to tak, ze ty checky dostavaj jen user konfiguraci + kontext, ve kterym to chces spoustet. Nic vic nechtej. Jediny co, ze ten kontext muzu nechat predat v `run()`

Reply:
Issue s "base test casem" lze vyřešit tak, že se použije trait místo base třídy. Pro tenhle případ ideal.

Neušetříme dva řádky, ale mnohem víc. Viz tvůj upravený example:

```php
<?php
use Cdn77\TestUtils\TestCheck\TestCheck;
use PHPUnit\Framework\TestCase;
/**
 * @group       integration
 * @testedClass none
 */
final class SuiteComplianceTest extends TestCaseBase
{
    use SuiteComplianceCheck;

    /** @return Generator<string, array{callable(self): TestCheck}> */
    public function providerChecks() : Generator
    {
        yield 'Test check title' => [new TestCheckClass(/* something TestCheck might need, nothing is required here */)],
        
        ...
    }
}
```

To je za mě výrazně lepší než to co máš teď, a nijak to uživatele neomezuje. Run metodu si píše sám, a dle interface dostane co může potřebovat. Konstruktor je taky kompletně jeho, takže si může přidat úplně cokoliv. Nic na úkor flexibility tu není.

> to nechapu, jaky veci tu jsou navic?
Právě samotný testCase + ty files. Není důvod nutit uživatele si kopírovat tenhle "get files" kód, a není třeba to dávat do konstruktoru. Můžeš to udělat bokem za usera, a checku to předávat v run metodě.

Reply:
takze jsme schovali metodu o dvou radcich do traity?

Reply:
v mém example toho chybí víc, než dva řádky

Reply:
jo ty se asi snazis smazat to predavani files, ale to ne. Point je, ze si to muzes pustit vickrat na ruzny sety files

Upravil jsem api, dalo mi to par napadu, ted je to sikovnejsi.

\+ jsem se konecne zbavil vsech trait a my je sem zas chceme davat abysme smazali run metodu. Check now, ale delat traitu, kvuli schovani metody je divny, no?

Reply:
Ok to pak chápu no, sice v tom nevidím přínos, ale chápu že je fajn tu možnost mít.

To API je teď lepší. Chce to teda ještě upravit readme, když už to není callable ale přímo ten TestCheck.

Taky by imo bylo lepší než to tu ukazovat v readme, tak takový test mít tady přímo v kódu, a v readme se na něj jako na example odkázat. WDYT?

Btw nemít traity nevidím jako nějakou výhodu, tady je to imo regulérní použití co to uživatelům zjednoduší, žádný zlo nebo něco nepřehlednýho. Prostě mi to přijde snažší na použití, elegantnější, a nijak to návrh/použití neomezuje.